exporters/exporter.go

```diff
@@ -0,0 +1,27 @@
+package exporters
+
+import "github.com/algorand/go-algorand/rpcs"
+
+// ExporterConfig is a generic type providing serialization/deserialization for exporter config files
+type ExporterConfig interface{}
+
+// Exporter defines the interface for plugins
+type Exporter interface {
+	// Connect will be called during initialization, before block data starts going through the pipeline.
+	// Typically used for things like initializing network connections.
+	// The ExporterConfig passed to Connect will contain the Unmarhsalled config file specific to this plugin.
+	// Should return an error if it fails--this will result in the Indexer process terminating.
+	Connect(cfg ExporterConfig) error
+
+	// Disconnect will be called during termination of the Indexer process.
+	// There is no guarantee that plugin lifecycle hooks will be invoked in any specific order in relation to one another.
+	// Returns an error if it fails which will be surfaced in the logs, but the process is already terminating.
+	Disconnect() error
+
+	// Receive is called for each block to be processed by the exporter.
+	// Should return an error on failure--retries are configurable.
+	Receive(blockData *rpcs.EncodedBlockCert) error
```

Comment:
This isn't quite right. Also, what do you think about wrapping it up in another object so that it's easier to add stuff without changing the function signature?

I'm also suggesting pass by value here, with the individual bits of data as pointers instead. We can make copies of the data object and allow plugins set each piece of data to nil if they want to (i.e. we are currently dropping the Certificate).

```
// ExportData is provided to the Exporter on each round.
type ExportData {
	// Block is the block data written to the blockchain.
	Block       *bookkeeping.Block

	// Delta contains a list of account changes resulting from the block. Processor plugins may have modify this data.
	Delta       *ledgercore.StateDelta

	// Certificate contains voting data that certifies the block. The certificate is non deterministic, a node stops collecting votes once the voting threshold is reached.
	Certificate *agreement.Certificate
}
```
```suggestion
	Receive(data ExportData) error
```